bin/app.rb

```diff
@@ -19,30 +20,56 @@ def list_books
 
   def list_people
     puts 'List of People:'
-    @people.each { |person| puts "ID: #{person.id} | Name: #{person.name} | Age: #{person.age}" }
+    @people.each { |person| display_person(person) }
+  end
+
+  def display_person(person)
+    if person.instance_of?(Teacher)
+      puts "[#{person.class}] ID: #{person.id} | Name: #{person.name} | Age: #{person.age} | " \
+           "Specialization: #{person.specialization}"
+    else
+      puts "[#{person.class}] ID: #{person.id} | Name: #{person.name} | Age: #{person.age} | Class: #{person.classroom}"
+    end
   end
 
   def create_person
     puts 'What type of person would you like to create? (1) Teacher (2) Student'
     type = gets.chomp.to_i
-    puts 'Enter the person name:'
-    name = gets.chomp
-    puts 'Enter the person age:'
-    age = gets.chomp.to_i
 
     case type
     when 1
-      puts 'Enter the teacher specialization:'
-      specialization = gets.chomp
-      person = Teacher.new(specialization, age, name: name)
+      create_teacher
     when 2
-      puts 'Enter the student classroom:'
-      classroom = gets.chomp
-      person = Student.new(Classroom.new(classroom), age, name: name)
+      create_student
+    else
+      puts 'Invalid input, Please enter a valid number!'
     end
 
```

Comment:
- [ ] Good job trying to fix the issue mentioned by the previous reviewer. However, I noticed while fixing the previous error, it is occurred the issue of creating a person of `type` `Student`. Right now, only `Teacher` can be successfully created. Make sure you fix that, below is the error I get while trying to create a `Student`
![Screenshot from 2023-03-30 07-44-47](https://user-images.githubusercontent.com/83042742/228827696-6bdf5d73-368c-47ba-89ea-c97fa999db6e.png)